pilot/pkg/xds/eds.go

```diff
@@ -551,7 +553,7 @@ func (eds *EdsGenerator) buildEndpoints(proxy *model.Proxy,
 	req *model.PushRequest,
 	w *model.WatchedResource) (model.Resources, model.XdsLogDetails) {
 	var edsUpdatedServices map[string]struct{}
-	if !req.Full {
+	if !req.Full || onlyEndpointsChanged(req) {
```

Comment:
This is only a special case, we donot push the unchanged eds. 

lgtm if the above theory is correct.

Reply:
Updated the PR with a flag in case we are wrong, and a comment explaining why its safe

Reply:
I think this is definitely safe since we already do it - "Full" is an istio internal concept, to envoy it sees the same thing -- only a subset of load assignments sent.

I am working on another change that is more extremely and that I am less certain is safe